analysisservice/src/main/java/nu/yona/server/analysis/service/AnalysisEngineService.java

```diff
@@ -132,53 +158,61 @@ private void analyze(ActivityPayload payload, Set<ActivityCategoryDTO> matchingA
 		// because the lock is per user, it doesn't matter much that we block early.
 		try (LockPool<UUID>.Lock lock = userAnonymizedSynchronizer.lock(payload.userAnonymized.getID()))
 		{
-			analyzeInsideLock(payload, matchingActivityCategories);
+			UserAnonymizedEntityHolder userAnonymizedHolder = new UserAnonymizedEntityHolder(payload.userAnonymized.getID());
+			analyzeInsideLock(userAnonymizedHolder, payload, matchingActivityCategories);
+			if (userAnonymizedHolder.isEntityFetched())
```

Comment:
The issue is that we neither know nor need to know whether the entity is dirty. If it was fetched, we should save it and leave it to Hibernate whether or not it needs to go to the database.